src/sleep/sleep.c

```diff
@@ -300,37 +305,54 @@ static int execute_s2h(const SleepConfig *sleep_config) {
                 r = fd_wait_for_event(tfd, POLLIN, 0);
                 if (r < 0)
                         return log_error_errno(r, "Error polling timerfd: %m");
+
                 if (!FLAGS_SET(r, POLLIN))  { /* We woke up before the alarm time, calculate battery discharged and estimate rate. */
-                        after_timestamp = now(CLOCK_MONOTONIC);
+                        after_timestamp = now(CLOCK_BOOTTIME);
 
-                        current_capacity = read_battery_capacity();
+                        time_elapsed = (after_timestamp - before_timestamp) / 1000000;
```

Comment:
USEC_PER_SEC instead of the number

Reply:
changed but still have to check if it is working fine. So will push after test. two tests pending,
- when battery is low then hibernation
- dis rate per hour is calculated correctly after using macro for microsec